smartlearner/optimizers/adam.py

```diff
@@ -0,0 +1,43 @@
+from collections import OrderedDict
+
+import theano
+import theano.tensor as T
+import numpy as np
+
+from ..interfaces import Optimizer
+from ..utils import sharedX
+
+class Adam(Optimizer):
```

Comment:
Why not subclass `SGD` like in `AdaGrad`? Oh wait, ok you really need the gradient. The opposite of the gradient cannot do since `mt` is signed, is that it?